framework/src/media/audio/audio_manager.c

```diff
@@ -161,15 +189,35 @@ audio_manager_result_t find_output_audio_card(void)
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
 		if ((type == 'p') && (card_id < CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+			struct audio_caps_desc_s caps_desc;
+			FAR struct audio_caps_s *caps = &(caps_desc.caps);
+			int fd = open(strcat(DEFAULT_AUDIO_DEVICE_PATH, dir_entry->d_name), O_WRONLY);
```

Comment:
DEFAULT_AUDIO_DEVICE_PATH is const char pointer. In this case you cannot use strcat.
This code occurs memory violation.
Make new char buffer to contain concat result, and use sprintf or sprintf_s instead.